### PR TITLE
Add notes/known issues section to installation page

### LINKED ISSUE #8306: Properly document that we have issues with multiple installed versions of OpenSSL.
Wehn users have both a distro installed version of OpenSSL and a manually installed local version, the build fails in strange and non-obvious ways.  We should document that we don't support building in such circumstances.

Long term I see low value in actually fixing the build for this, as it is going to become a much less common situation as OpenSSL 1.1 becomes more and more common.

### PR DESCRIPTION
**Pull request by joelhans**
##### Summary

Fixes #8306 

I added this notes section to the installation page where we can put notices like SSL and workarounds for various distros. I don't love cluttering up that page any more, but I can't figure out where else we could put this information with the best mix of discoverability and not being too in-your-face.

##### Component Name

packaging/

### CODE REVIEW

**Comment by Ferroin** [comment]
We might want to mention here that the build also currently does not work with LibreSSL in place of OpenSSL, and also fails when using Clang in some configurations.

**Comment by joelhans** [comment]
@Ferroin Done!

File: packaging/installer/README.md [hunk 1]
```diff
@@ -165,3 +165,23 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+## Installation notes
+
+We are tracking a few ongoing issues related to installation and packaging.
+
+**Multiple versions of SSL**: We've received reports from the community about issues with running the
+`kickstart.sh` script on systems that have both a distribution-installed version of OpenSSL and a manually-installed
+local version. The Agent's installer cannot handle both, and so you must remove one or the other to install the Agent.
+
+**LibreSSL**: The Agent installer is only compatible with OpenSSL. Critical functions do not work on systems with
+LibreSSL installed as a replacement for OpenSSL.
```
Comment (Ferroin) [inline packaging/installer/README.md:178]: Actually, the build fails completely if you try to use LibreSSL instead of OpenSSL. They use the same naming for libraries and headers but aren't API compatible.

Comment (thiagoftsm) [reply packaging/installer/README.md:178]: @Ferroin I am working on a new PR with TLS right now, I will bring a fix for this on it.

Comment (joelhans) [reply packaging/installer/README.md]: @Ferroin Done.

@thiagoftsm We can always edit this text again when your PR is ready.

**Comment by prologic** [comment]
> We might want to mention here that the build also currently does not work with LibreSSL in place of OpenSSL, and also fails when using Clang in some configurations.

There was a PR to fix this. I'm not sure this is true anymore. Let me find the PR.

**Comment by prologic** [comment]
Issue tracking LibreSSL fix #8796

**Comment by prologic** [comment]
We should hold of fon merging this until #8796 is resolved otherwise it makes us look a bit silly to say it doens't compile with LibreSSL one day but does the next. :)

**Comment by thiagoftsm** [comment]
@prologic the PR #9068 will fix the problem when it is merged, this is the motive I closed #8796.

**Comment by joelhans** [comment]
> We should hold of fon merging this until #8796 is resolved otherwise it makes us look a bit silly to say it doens't compile with LibreSSL one day but does the next. :)

Waiting is fine with me. You guys can ping me when this is ready for an update and I'll get on it.

**Comment by joelhans** [comment]
An update on this one: @zack-shoylev and I need to add a few more workarounds for old systems into this PR, so I changed the title to better reflect what we're doing and we'll ask for reviews again when we've added all the notes.

Context is at netdata/marketing#240.

**Comment by zack-shoylev** [comment]
Should we add @ktsaou's notes about running on ubuntu 14 from https://github.com/netdata/netdata/issues/8905 ? will have to add some security warnings (essentially you are disabling security so a hostile intermediary could steal data you are uploading such as metrics).

**Comment by prologic** [comment]
Probably a good idea. Should also mention that using the Static install is a valid solution here (_not even a work-around either_).

**Comment by joelhans** [comment]
@zack-shoylev Of course—that's been the plan all along. I'm hoping to get it done today.

File: packaging/installer/README.md [hunk 2]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+> Allowing self-signed certificates and skipping hostname checks could allow a hostile intermediary to collect your
+> node's metrics as it transits to Netdata Cloud.
+
+If you choose to proceed, you must export the `-DACLK_SSL_ALLOW_SELF_SIGNED` CFLAG before installation.
+
+```bash
+export CFLAGS="-DACLK_SSL_ALLOW_SELF_SIGNED"
+```
+
+Then install Netdata using your method of choice.
```
Comment (zack-shoylev) [inline packaging/installer/README.md:197]: > Note that this will leave your Netdata Cloud connection vulnerable to man-in-the-middle attacks and is not recommended.

File: claim/README.md [hunk 3]
```diff
@@ -186,6 +186,15 @@ might be having with the ACLK or claiming process.
...
 Use these keys and the information below to troubleshoot the ACLK.
 
+#### Claiming on older distributions (CentOS 6/7, Debian 8, Ubuntu 14.04)
+
+If you're running an older Linux distribution or one that has reached EOL, such as Ubuntu 14.04 LTS, Debian 8, or CentOS
+6/7, your Agent may not be able to connect to Cloud.
+
+You should either reinstall Netdata with a [static build](/packaging/installer/methods/kickstart-64.md), or see the
+[known installation issues](/packaging/installer/README.md#failure-to-claim-connect-to-cloud) for an OpenSSL workaround
+that comes at the cost of decreased security.
```
Comment (prologic) [inline claim/README.md:196]: Nit: This is __NOT__ a "work-around" and AFAIk effectively removes __ALL_ security. Without host verification you have absolutely zero idea who/what you are connecting to.

File: packaging/installer/README.md [hunk 4]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+### Failure to claim/connect to Netdata Cloud
+
+If you're running an older Linux distribution or one that has reached EOL, such as Ubuntu 14.04 LTS, Debian 8, or CentOS
+6/7, your Agent may not be able to connect to Cloud due to an old version of OpenSSL that is incompatible with Netdata
+Cloud.
+
+We recommend using the [static build installation](/packaging/installer/methods/kickstart-64.md) to avoid this issue.
+You can, however, use the workaround below.
+
+#### OpenSSL workaround
```
Comment (prologic) [inline packaging/installer/README.md:182]: Again Nit: Not a work-around.

File: packaging/installer/README.md [hunk 5]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+
+Instead of using the static build, you can also reconfigure OpenSSL to allow self-signed certificates and skip hostname
```
Comment (prologic) [inline packaging/installer/README.md:184]: @amoss / @netdata/infra Please correct me if I'm wrong here, but I don't think "self-signed" certs are really relevant here. It is simply that we are disabling hostname verification.

Comment (prologic) [reply packaging/installer/README.md:184]: In fact I'm kind of curious; @amoss Do we require all 3 bit flags here to make this work on systems with an older openssl that do not have `X509_VERIFY_PARAM_set1_host` feature

```#!c
LCCSCF_ALLOW_SELFSIGNED | LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK | LCCSCF_ALLOW_INSECURE
```

Is it not enough to set `LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK` option?

File: packaging/installer/README.md (hunk 2 above)
Comment (prologic) [reply packaging/installer/README.md:197]: I _think_ we need to review this very carefully...

File: packaging/installer/README.md [hunk 6]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+To use the `CFLAGS` workaround on Ubuntu 14.04 LTS, you must first install `libuv1-dev` via a PPA:
+
+```bash
+add-apt-repository ppa:acooks/libwebsockets6
+apt-get update
+apt-get install libuv1-dev
+```
+
+Then proceed with the CFLAGS workaround described above.
+
```
Comment (prologic) [inline packaging/installer/README.md:210]: This is already handled by #8916 not sure we need to duplicate the code in our docuemtnation as we don't do this for similar reasons for handling other systems.

File: packaging/installer/README.md [hunk 7]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+### CentOS 6 and CentOS 8
+
+To install the Agent on certain CentOS and RHEL systems, you must enable non-default repositories, such as EPEL or
+PowerTools, to gather hard dependencies. See the [CentOS 6](/packaging/installer/methods/manual.md#centos-rehel-6-x) and
+[CentOS 8](/packaging/installer/methods/manual.md#centos-rehel-8-x) sections for more information.
```
Comment (prologic) [inline packaging/installer/README.md:215]: Why is this here? The installer takes care of this, it even tells the user and prompts then (_optionally_).

File: packaging/installer/README.md [hunk 8]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+
+### Multiple versions of OpenSSL
+
+We've received reports from the community about issues with running the `kickstart.sh` script on systems that have both
+a distribution-installed version of OpenSSL and a manually-installed local version. The Agent's installer cannot handle
+both, and so you must remove one or the other to install the Agent.
```
Comment (prologic) [inline packaging/installer/README.md:221]: Alternatively use the Static Netdata.

File: packaging/installer/README.md [hunk 9]
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
...
+
+### Clang compiler on Linux
+
+Our current build process has some issues when using certain configurations of the `clang` C compiler on Linux. See [the
+section on `nonrepresentable section on output`
+errors](/packaging/installer/methods/manual.md#nonrepresentable-section-on-output-errors) for a workaround.
```
Comment (prologic) [inline packaging/installer/README.md:227]: These are actively being fixed and will no longer be an issue very soon.

File: packaging/installer/README.md (hunk 5 above)
Comment (knatsakis) [reply packaging/installer/README.md:184]: @prologic I concur with both you comments:

> I don't think "self-signed" certs are really relevant here. It is simply that we are disabling hostname verification.

￼I thought so too.

> Is it not enough to set LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK option?

My understanding is that it should be enough.

Comment (amoss) [reply packaging/installer/README.md:184]: I think that it would be enough.

Comment (prologic) [reply packaging/installer/README.md:184]: Then I would prefer we remove the wording around "self-signed" and "work-around" and state what's actually going on here.

**Comment by joelhans** [comment]
[Help Zack with docs on running Netdata on old systems](https://trello.com/c/QLKDYINS/54-help-zack-with-docs-on-running-netdata-on-old-systems)

**Comment by joelhans** [comment]
[Help Zack with docs on running Netdata on old systems](https://trello.com/c/QLKDYINS/54-help-zack-with-docs-on-running-netdata-on-old-systems)

**Comment by joelhans** [comment]
@prologic Given that I wrote this PR on Friday and #9198 was then merged over the weekend, @zack-shoylev and I will need to take some time to figure out how to redesign this entire PR to accommodate those changes. More soon.

File: packaging/installer/README.md (hunk 7 above)
Comment (joelhans) [reply packaging/installer/README.md:215]: This is here because of the linked sections ([6](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/manual/#centos--rhel-6x) and [8](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/manual/#centos--rhel-6x)), which were added some months ago to explain how to install on these distros, since users were running into issues.

If those sections are no longer needed, let me know and I'll remove them in a separate PR.

File: packaging/installer/README.md (hunk 6 above)
Comment (joelhans) [reply packaging/installer/README.md:210]: That PR is not merged yet. Is some other fix already present for Ubuntu 14.04?

File: packaging/installer/README.md (hunk 5 above)
Comment (joelhans) [reply packaging/installer/README.md:184]: The self-signed stuff was my bad—I misinterpreted what the option was doing.

**Comment by prologic** [comment]
Thanks @joelhans :) Let me know if you need any help!

**Comment by joelhans** [comment]
@prologic This is actually ready for another review when you have time. I chatted with @zack-shoylev off GitHub about what's in the PR right now, and while he's happy with it, the opinions of you and @amoss are what really count!

File: packaging/installer/README.md (hunk 4 above)
Comment (zack-shoylev) [reply packaging/installer/README.md]: I'm fine with removing the work-around wording, it's good to make it clear.

Do note that users running on old machines they are most likely (hopefully) just using for testing Netdata won't care they are disabling security.

File: packaging/installer/README.md (hunk 7 above)
Comment (prologic) [reply packaging/installer/README.md:191]: I would just add a note to say that our installer automatically takes cares of this if you use our `kickstart.sh`.

**Comment by joelhans** [comment]
@zack-shoylev Can you hop in here and make any last requests or approve? I think merging is blocked, in part, because you requested changes. Thanks!

**Comment by prologic** [comment]
You had two approvals. Merged.
